misc/bazel/csharp.bzl

```
@@ -0,0 +1,73 @@
+load("@rules_dotnet//dotnet:defs.bzl", "csharp_binary", "csharp_library", "csharp_test", "publish_binary")
+load("@rules_pkg//pkg:mappings.bzl", "strip_prefix")
+load("@semmle_code//:dist.bzl", "pack_zip")
+load("//:defs.bzl", "codeql_platform")
+
+TARGET_FRAMEWORK = "net8.0"
+
+def codeql_csharp_library(name, **kwargs):
+    nullable = kwargs.pop("nullable", "enable")
+    target_frameworks = kwargs.pop("target_frameworks", [TARGET_FRAMEWORK])
```

why pop defaults from `**kwargs` instead of just listing these in the default arguments?

While I remember commenting on this in previous reviews, starlark doesn't suffer from the python gotcha of list default arguments, so we can actually use those safely.

---

Yeah I think this is a common python-ism, that we also adhere to in other parts of our codebase (probably not to induce the mental overhead of: "is this safe?" every tiem you see it).

---

if we want to go this way, we could do this to limit the following boilerplate of explicitly passing the singled-out parameters:
```suggestion
    kwargs.setdefault("nullable", "enable")
    kwargs.setdefault("target_frameworks", [TARGET_FRAMEWORK])
```